core/src/main/java/org/apache/iceberg/actions/rewrite/BinPackStrategy.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions.rewrite;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.actions.RewriteDataFiles;
+import org.apache.iceberg.actions.RewriteDataFiles.Strategy;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.util.BinPacking;
+import org.apache.iceberg.util.BinPacking.ListPacker;
+import org.apache.iceberg.util.PropertyUtil;
+
+abstract class BinPackStrategy implements RewriteStrategy {
+
+  /**
+   * Minimum number of files that need to be in a file group to be considered
+   * for rewriting.
+   */
+  public static final String MIN_NUM_INPUT_FILES = "min-num-input-files";
```

Comment:
why do these constants need to be public? alternatively, should these options be defined in the `RewrtieStrategy`? are they specific to `BinPackStrategy`?

Reply:
These are constants for end users to use, so they should be public. In the discussion on the RewriteStrategy we decided they were bin-pack specific since the other "sort" strategy will probably be filtering files based on clustering rather than the number of files that are a specific size

Reply:
Is `num` implied? Files is the unit here, so I think we can probably drop it.

Reply:
I also think we can drop `num`.

Reply:
Sure, I was using this ordering because it was easier for me to see the properties and related variables but I can swap it to all the statics first

Reply:
Resolving this as it has been discussed and addressed.